### DISCUSSION TITLE
MX Vulkan

### DISCUSSION THREAD

**Discussion by A1D2D** [Dav1dde/glad#485] [Q&A] [answered]
Hey i'm having problems with setting up multi context vulkan. I found gladLoadVulkanContext to create a VKContext but it needs a VkPhysicalDevice as a input how can i get one or how can i create a GladVulkanContext ?

**Comment by Dav1dde** [comment]
You load without a physical device (`NULL`) and later once you've acquired a device you load again but this time with the device. You can follow the [vulkan example](https://github.com/Dav1dde/glad/blob/glad2/example/c/vulkan_tri_glfw/vulkan_tri_glfw.c) just use the context API's instead.

**Reply by A1D2D** [reply]
Ty I tried already with a null device, should gladLoadVulkanContext() return a 0 if there is a null device, or i just messed up something else?

**Reply by Dav1dde** [reply, marked as answer]
It should not return `0`, something else is wrong.

**Reply by A1D2D** [reply]
i use the same getProcAdress function that i give to my glad opengl version, so i dont know if there is a difference in the load function between the two APis and i assume for the first arg i just need to pass in a pointer to a GladVulkanContext. 
```
GladVulkanContext tc;
 if(!gladLoadVulkanContext(&tc, nullptr, MTWM_Win::getProcAddress)) {
    std::cerr << "failed to load temp\n";
 }
``` 
//MTWM_Win::getProcAddress works fine with gladLoadGLContext()

**Reply by A1D2D** [reply]
Nah sry i messed up, i need a different getProcFunction for vulkan sry
